src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,319 @@
+        }
+    }
+
+    private static void removeTaskFromList(ArrayList<Task> tasks, String s) {
+        int taskNumber = Integer.parseInt(s)-1;
+        Task taskToRemove = tasks.get(taskNumber);
+        tasks.remove(taskNumber);
+        Task.reduceTaskCount();
+        printDeleteTaskSuccess(taskToRemove);
+    }
+
+    private static void saveTasksToFile(ArrayList<Task> tasks) {
+        final String fileName = "data/TaskList.txt";
+        final String directoryName = "data/";
+        File taskFile = new File(fileName);
+        File directoryFile = new File(directoryName);
+
+        if (directoryFile.mkdir()) {
+            printLinesWithText("Directory to store save file created");
+        }
+        try {
+            if (taskFile.createNewFile()){
+               printLinesWithText("Save file for tasks created");
+            }
+            FileWriter fileWriter = new FileWriter(fileName);
+            writeToFile(tasks, fileWriter);
+            fileWriter.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred - Writing");
+        }
+    }
+
+    private static void writeToFile(ArrayList<Task> tasks, FileWriter fileWriter) throws IOException {
+        for (Task task : tasks) {
+            String textToAdd = generateTextToAddString(task);
+            fileWriter.write(textToAdd + System.lineSeparator());
+        }
+    }
+
+    private static String generateTextToAddString(Task task) {
+        switch(task.getType()) {
+        case 'T':
+            return task.getType() + " | " + task.isDone() + " | " + task.getDescription();
+        case 'D':
+            Deadline d = (Deadline) task;
+            return d.getType() + " | " + d.isDone() + " | " + d.getDescription() + " | " + d.getBy();
+        case 'E':
+            Event e = (Event) task;
+            return e.getType() + " | " + e.isDone() + " | " + e.getDescription() + " | " + e.getAt();
+        default:
+            return "Unrecognized type!";
+        }
+    }
+
+    private static void loadSaveData(ArrayList<Task> tasks) {
+        String fileName = "data/TaskList.txt";
+        File taskFile = new File(fileName);
+
+        try {
+            if (taskFile.exists()) {
+                Scanner reader = new Scanner(taskFile);
+                readSaveData(tasks, reader);
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred");
+        }
+    }
+
+    private static void addSavedTask(ArrayList<Task> tasks, Task t, String status) {
+        if (status.equals("true")) {
+            t.setDone();
+        }
+        tasks.add(t);
+    }
+
+    private static void readSaveData(ArrayList<Task> tasks, Scanner reader) {
+        while (reader.hasNextLine()) {
+            String data = reader.nextLine();
+            String[] taskInfoArray = data.split(" \\| ");
+
+            switch (taskInfoArray[0]) {
+            case "T":
+                ToDo t = new ToDo(taskInfoArray[2]);
```

Comment:
Even though the scope is small, you can consider giving a more descriptive name to `t`.